pylint/checkers/base.py

```diff
@@ -203,7 +203,7 @@ def _has_abstract_methods(node):
     The methods should be made abstract by decorating them
     with `abc` decorators.
     """
-    return len(utils.unimplemented_abstract_methods(node)) > 0
+    return bool(utils.unimplemented_abstract_methods(node))
```

Comment:
I've made a small modification to the existing code. It will traverse the parent nodes until it reaches a node like `if` or `while` where the trithiness of the value is checked. It should be fine now.